projects/plugins/protect/src/js/components/scan-onboarding/use-scan-popover-args.jsx

```diff
@@ -0,0 +1,142 @@
+import { Text, Button, getRedirectUrl, useBreakpointMatch } from '@automattic/jetpack-components';
+import { useProductCheckoutWorkflow } from '@automattic/jetpack-connection';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import API from '../../api';
+import { JETPACK_SCAN_SLUG } from '../../constants';
+import useAnalyticsTracks from '../../hooks/use-analytics-tracks';
+import useOnboarding from '../../hooks/use-onboarding';
+
+const useScanPopoverArgs = ( { anchors, totalSteps } ) => {
+	const { adminUrl, siteSuffix } = window.jetpackProtectInitialState;
+	const { incrementOnboardingStep, dismissOnboarding, createPopoverArgs } = useOnboarding();
+	const [ isSm ] = useBreakpointMatch( 'sm' );
+	const { run } = useProductCheckoutWorkflow( {
+		productSlug: JETPACK_SCAN_SLUG,
+		redirectUrl: adminUrl,
+	} );
+	const { recordEventHandler } = useAnalyticsTracks();
+	const getScan = recordEventHandler( 'jetpack_protect_onboarding_get_scan_link_click', run );
+
+	const handleDismiss = () => {
+		API.scanOnboardingDismissed();
+	};
+
+	return {
+		yourScanResults: createPopoverArgs( {
```

Comment:
Why is the `createPopoverArgs` pattern necessary here? If it is to inject data from the `useOnboarding` hook, could those properties be exported directly from the hook?

Reply:
This could probably be done better - `createPopArgs` is meant to take the unique args for each popover instance and apply all args that are shared. This could likely just be used a single time once the current popover args are identified before they are passed to `ActionPopover` component.